pkg/app/piped/cloudprovider/ecs/task_test.go

```diff
@@ -78,3 +80,205 @@ cpu: 256
+		},
+		{
+			name: "missing containerDefinitions",
+			input: []byte(`
+{
+	"family": "nginx-canary-fam-1",
+	"compatibilities": [
+		"FARGATE"
+	],
+	"networkMode": "awsvpc",
+	"memory": 512,
+	"cpu": 256,
+}
+`),
+			expected:    nil,
+			expectedErr: true,
+		},
+		{
+			name: "missing image name",
+			input: []byte(`
+{
+	"family": "nginx-canary-fam-1",
+	"compatibilities": [
+		"FARGATE"
+	],
+	"networkMode": "awsvpc",
+	"memory": 512,
+	"cpu": 256,
+	"containerDefinitions" : [
+		{
+			"image": "gcr.io/pipecd/:v1.0.0",
+			"name": "helloworld",
+			"portMappings": [ 
+				{ 
+				"containerPort": 80,
+				"hostPort": 9085,
+				"protocol": "tcp"
+				}
+			]
+		}
+	]
+}
+`),
+			expected:    nil,
+			expectedErr: true,
+		},
+		{
+			name: "multiple containers",
+			input: []byte(`
+{
+	"family": "nginx-canary-fam-1",
+	"compatibilities": [
+		"FARGATE"
+	],
+	"networkMode": "awsvpc",
+	"memory": 512,
+	"cpu": 256,
+	"containerDefinitions" : [
+		{
+			"image": "gcr.io/pipecd/helloworld:v1.0.0",
+			"name": "helloworld",
+			"portMappings": [ 
+				{ 
+				"containerPort": 80,
+				"hostPort": 9085,
+				"protocol": "tcp"
+				}
+			]
+		},
+		{
+			"image": "gcr.io/pipecd/my-service:v1.0.0",
+			"name": "my-service",
+			"portMappings": [ 
+				{ 
+				"containerPort": 80,
+				"hostPort": 9090,
+				"protocol": "tcp"
+				}
+			]
+		}
+	]
+}
+`),
+			expected: []*model.ArtifactVersion{
+				{
+					Kind:    model.ArtifactVersion_CONTAINER_IMAGE,
+					Version: "v1.0.0",
+					Name:    "helloworld",
+					Url:     "gcr.io/pipecd/helloworld:v1.0.0",
+				},
+				{
+					Kind:    model.ArtifactVersion_CONTAINER_IMAGE,
+					Version: "v1.0.0",
+					Name:    "my-service",
+					Url:     "gcr.io/pipecd/my-service:v1.0.0",
+				},
+			},
+			expectedErr: false,
+		},
+		{
+			name: "multiple containers with the same image",
+			input: []byte(`
+{
+	"family": "nginx-canary-fam-1",
+	"compatibilities": [
+		"FARGATE"
+	],
+	"networkMode": "awsvpc",
+	"memory": 512,
+	"cpu": 256,
+	"containerDefinitions" : [
+		{
+			"image": "gcr.io/pipecd/helloworld:v1.0.0",
+			"name": "helloworld",
+			"portMappings": [ 
+				{ 
+				"containerPort": 80,
+				"hostPort": 9085,
+				"protocol": "tcp"
+				}
+			]
+		},
+		{
+			"image": "gcr.io/pipecd/helloworld:v1.0.0",
+			"name": "helloworld-02",
+			"portMappings": [ 
+				{ 
+				"containerPort": 80,
+				"hostPort": 9091,
+				"protocol": "tcp"
+				}
+			]
+		}
+	]
+}
+`),
+			expected: []*model.ArtifactVersion{
+				{
+					Kind:    model.ArtifactVersion_CONTAINER_IMAGE,
+					Version: "v1.0.0",
+					Name:    "helloworld",
+					Url:     "gcr.io/pipecd/helloworld:v1.0.0",
+				},
+			},
+			expectedErr: false,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			td, _ := parseTaskDefinition(tc.input)
+			versions, err := FindArtifactVersions(td)
+			assert.Equal(t, tc.expectedErr, err != nil)
+			assert.Equal(t, tc.expected, versions)
```

Comment:
```suggestion
			assert.ElementsMatch(t, tc.expected, versions)
```
should use this to avoid order of results while checking